design/open/spark-outputcommitter/perf.md

```diff
@@ -0,0 +1,173 @@
+# Performance issues with GC
+
+## About
+
+This is a short summary of observed performance issues as of 2022-11-10.
+Things are changing!
+
+## The "Amazon reviews" test
+
+I recreated the XXXX tests:
+
+```scala
+// import data from public data set
+val AmazonReviews = spark.read.parquet("s3://amazon-reviews-pds/parquet/")
+val partial = AmazonReviews.limit(117647)
+partial.write.parquet("lakefs://test-1/main/amazon-reviews")
+```
+
+Now I run on a clean branch:
+
+```scala
+val reviews = spark.read.parquet("lakefs://test-1/main/amazon-reviews")
+
+reviews.repartition(4000)
+	.write
+	.csv("lakefs://test-1/test-clean-2/amazon-reviews-repartition-clean2.csv/")
+```
+
+Notebook available to Treeversers on request.
+
+I repeated the tests with PostgreSQL and DynamoDB, and with 16, 3 and a
+single worker.  I did not cover the entire matrix.
+
+This will perform 4000 merges concurrently across all the workers, followed
+by a single uncontended merge.  Each of the 4000 merges adds a *single*
+file, and the repo is almost empty.  The last merge adds 4001 files.
```

Comment:
Just to verify - each file that was produced ~12GB? or bigger (the source is compressed parquet)

Reply:
No, not at all.  The total file size is 42.4 MiB (Parquet, so compressed).  The job splits it up into 4000 partitions, each is 15-30 KiB (CSV, so uncompressed).

The point of the test is to stress the metadata part of lakeFS.  Actual data is handled by S3 directly, so less point in stressing that.  We want to see a limiting case for 4000 partitions.